samples/debugmalloc/README.md

```diff
@@ -0,0 +1,328 @@
+// sdk tool oeedger8r against the debugmalloc.edl file.
+#include "debugmalloc_t.h"
+
+// This is the function that the host calls. It prints
+// a message in the enclave before calling back out to
+// the host to print a message from there too.
+void enclave_hello()
+{
+    void *buf1, *buf2, *buf3;
+    uint64_t num;
+    char* report;
+
+    oe_assert(oe_debug_malloc_tracking_start() == OE_OK);
+    buf1 = oe_malloc(4096);
+    oe_assert(oe_debug_malloc_tracking_stop() == OE_OK);
+    if (oe_debug_malloc_tracking_report(&num, &report) == OE_OK)
+    {
+        printf("There are %d un-freed objects reported:\n%s\n", num, report);
+        oe_free(report);
+    }
+
+    oe_assert(oe_debug_malloc_tracking_start() == OE_OK);
+    buf2 = oe_malloc(4096);
+    oe_assert(oe_debug_malloc_tracking_stop() == OE_OK);
+
+    oe_assert(oe_debug_malloc_tracking_start() == OE_OK);
+    buf3 = oe_malloc(4096);
+    oe_assert(oe_debug_malloc_tracking_stop() == OE_OK);
+    if (oe_debug_malloc_tracking_report(&num, &report) == OE_OK)
+    {
+        printf("There are %d un-freed objects reported:\n%s\n", num, report);
+        oe_free(report);
+    }
+
+    oe_free(buf1);
+    oe_free(buf2);
+    oe_free(buf3);
+
+    // Call back into the host
+    oe_result_t result = host_hello();
+    if (result != OE_OK)
+    {
+        fprintf(
+            stderr,
+            "Call to host_hello failed: result=%u (%s)\n",
+            result,
+            oe_result_str(result));
+    }
+}
+```
+Please make sure the report is released after each calling of
+oe_debug_malloc_tracking_report(), since it incurs allocation at every turn.
+
+For the local detection, the expected output result are 1 and 3 leaks for each
+report respectively:
+```
+There are 1 un-freed objects reported:
+oe_debug_malloc(): 7efc950f1f90
+oe_malloc(): 7efc950f3463
+enclave_hello(): 7efc9501804d
+ecall_enclave_hello(): 7efc95003762
+oe_handle_call_enclave_function(): 7efc95126cd9
+_handle_ecall(): 7efc95128bae
+__oe_handle_main(): 7efc951284ef
+oe_enter(): 7efc95129e6c
+
+
+There are 3 un-freed objects reported:
+oe_debug_malloc(): 7efc950f1f90
+oe_malloc(): 7efc950f3463
+enclave_hello(): 7efc950183d6
+ecall_enclave_hello(): 7efc95003762
+oe_handle_call_enclave_function(): 7efc95126cd9
+_handle_ecall(): 7efc95128bae
+__oe_handle_main(): 7efc951284ef
+oe_enter(): 7efc95129e6c
+
+oe_debug_malloc(): 7efc950f1f90
+oe_malloc(): 7efc950f3463
+enclave_hello(): 7efc95018265
+ecall_enclave_hello(): 7efc95003762
+oe_handle_call_enclave_function(): 7efc95126cd9
+_handle_ecall(): 7efc95128bae
+__oe_handle_main(): 7efc951284ef
+oe_enter(): 7efc95129e6c
+
+oe_debug_malloc(): 7efc950f1f90
+oe_malloc(): 7efc950f3463
+enclave_hello(): 7efc9501804d
+ecall_enclave_hello(): 7efc95003762
+oe_handle_call_enclave_function(): 7efc95126cd9
+_handle_ecall(): 7efc95128bae
+__oe_handle_main(): 7efc951284ef
+oe_enter(): 7efc95129e6c
+
+```
+
+For the global detection, Since this sample has no memory leak at the point of
```

Comment:
It would be good to demonstrate detection as well.
- The host can allocate the same enclave again.
- The host makes an ecall that allocates some memory, but does not free it
- Host calls oe_terminate on the enclave which will now return OE_LEAK

Reply:
Thanks for the suggestion, I update the doc to let users to try it out.